google-cloud-datastore/src/test/java/com/google/cloud/datastore/execution/AggregationQueryExecutorTest.java

```diff
@@ -0,0 +1,177 @@
+  public void shouldExecuteAggregationQuery() {
+    EntityQuery nestedQuery =
+        Query.newEntityQueryBuilder()
+            .setNamespace(NAMESPACE)
+            .setKind(KIND)
+            .setFilter(eq("done", true))
+            .build();
+
+    AggregationQuery aggregationQuery =
+        Query.newAggregationQueryBuilder()
+            .setNamespace(NAMESPACE)
+            .addAggregation(count().as("total"))
+            .over(nestedQuery)
+            .build();
+
+    RunAggregationQueryResponse runAggregationQueryResponse = dummyAggregationQueryResponse();
+    expect(mockRpc.runAggregationQuery(anyObject(RunAggregationQueryRequest.class)))
+        .andReturn(runAggregationQueryResponse);
+
+    replay(mockRpc);
+
+    AggregationResults aggregationResults = queryExecutor.execute(aggregationQuery);
+
+    verify(mockRpc);
+    assertThat(aggregationResults)
+        .isEqualTo(
+            new AggregationResults(
+                asList(
+                    new AggregationResult(
+                        ImmutableMap.of(
+                            "count", LongValue.of(209), "property_2", LongValue.of(100))),
+                    new AggregationResult(
+                        ImmutableMap.of(
+                            "count", LongValue.of(509), "property_2", LongValue.of(100)))),
+                Timestamp.fromProto(runAggregationQueryResponse.getBatch().getReadTime())));
+  }
+
+  @Test
+  public void shouldExecuteAggregationQueryWithReadOptions() {
+    EntityQuery nestedQuery =
+        Query.newEntityQueryBuilder()
+            .setNamespace(NAMESPACE)
+            .setKind(KIND)
+            .setFilter(eq("done", true))
+            .build();
+
+    AggregationQuery aggregationQuery =
+        Query.newAggregationQueryBuilder()
+            .setNamespace(NAMESPACE)
+            .addAggregation(count().as("total"))
+            .over(nestedQuery)
+            .build();
+
+    RunAggregationQueryResponse runAggregationQueryResponse = dummyAggregationQueryResponse();
+    expect(mockRpc.runAggregationQuery(matches(runAggregationRequestWithEventualConsistency())))
+        .andReturn(runAggregationQueryResponse);
+
+    replay(mockRpc);
+
+    AggregationResults aggregationResults =
+        queryExecutor.execute(aggregationQuery, eventualConsistency());
+
+    verify(mockRpc);
+    assertThat(aggregationResults)
+        .isEqualTo(
+            new AggregationResults(
+                asList(
+                    new AggregationResult(
+                        ImmutableMap.of(
+                            "count", LongValue.of(209), "property_2", LongValue.of(100))),
+                    new AggregationResult(
+                        ImmutableMap.of(
+                            "count", LongValue.of(509), "property_2", LongValue.of(100)))),
+                Timestamp.fromProto(runAggregationQueryResponse.getBatch().getReadTime())));
+  }
+
+  private RunAggregationQueryResponse dummyAggregationQueryResponse() {
```

Comment:
rename to `placeholderAggregationQueryResponse`

Reply:
Fixed as part of https://github.com/googleapis/java-datastore/pull/823/commits/a5b462332db2f8b88db4016d2f83046c89e74acd.